ruby_event_store/lib/ruby_event_store/spec/event_repository_lint.rb

```diff
@@ -947,4 +947,8 @@ def self.new(
     end
   end
 
+  specify 'read returns enumerator' do
+    specification = RubyEventStore::Specification.new(repository)
+    expect(repository.read(specification.result)).to be_kind_of(Enumerator)
```

Comment:
If the repository decided to be eager-loading instead of lazy-loading that's ok for me. I would expect `Enumerable`, not `Enumerator`.

Reply:
@pawelpacana I was wrong and right at the same time :)

if the Specification does:

```
def each
  @repository.read(result)
```

like it is doing right now, then we would need an `Enumerator`. Otherwise we would need to do `client.read.each.each do` because we are not passing a block

If the Specification did:

```
def each(&block)
  @repository.read(result).each(&block)
```

then we would need an `Enumerable` (and `Enumerator` is `Enumerable` but so is `Array`) and I think it makes most sense.

the Specification could also do:

```
def each(&block)
  @repository.read(result, &block)
```

but I don't like it.

Reply:
Seprate topic: Should we include Enumerable to Specification?

Reply:
@paneq I had thought about including `Enumerable` in `Specification` too....